src/terrain/terrain_base.cpp

```diff
@@ -167,12 +167,12 @@ TerrainBase::stamp_tile_region(
     TerrainOffset x_end = stamp.x_end + x_offset * area_size_ + area_size_ / 2;
     TerrainOffset y_end = stamp.y_end + y_offset * area_size_ + area_size_ / 2;
 
-    if (stamp.elements_can_stamp.has_value()) {
-        for (TerrainOffset x = x_start; x < x_end; x++) {
-            for (TerrainOffset y = y_start; y < y_end; y++) {
-                for (TerrainOffset z = stamp.z_start; z < stamp.z_end; z++) {
-                    if (in_range(x, y, z)) {
-                        Tile* tile = get_tile(x, y, z);
+    for (TerrainOffset x = x_start; x < x_end; x++) {
+        for (TerrainOffset y = y_start; y < y_end; y++) {
+            for (TerrainOffset z = stamp.z_start; z < stamp.z_end; z++) {
+                if (in_range(x, y, z)) {
+                    Tile* tile = get_tile(x, y, z);
+                    if (stamp.elements_can_stamp.has_value()) {
                         if (has_tile_material(stamp.elements_can_stamp.value(), tile)) {
```

Comment:
this should be or not and

also

ditch the optional and just check if the set is empty. no need for another level of indirection

Reply:
no

Reply:
I don't want to create an entire set just to remove one tab.

Reply:
fine